src/librustc_mir/borrow_check/places_conflict.rs

```diff
@@ -29,281 +31,160 @@ pub(super) fn places_conflict<'gcx, 'tcx>(
         borrow_place, access_place, access
     );
 
-    unroll_place(borrow_place, None, |borrow_components| {
-        unroll_place(access_place, None, |access_components| {
-            place_components_conflict(tcx, mir, borrow_components, access_components, access)
-        })
-    })
-}
+    let place_elements_conflict = |tcx: TyCtxt<'_, 'gcx, 'tcx>,
```

Comment:
So, stylistically, I would prefer to keep this as a separate function. I really dislike large closures -- it's hard to distinguish what state they access from the surrounding environment.